src/libstd/keyword_docs.rs

```diff
@@ -558,6 +558,28 @@ mod impl_keyword { }
 /// [Reference]: ../reference/statements.html#let-statements
 mod let_keyword { }
 
+#[doc(keyword = "while")]
+//
+/// while loop
+///
+/// `while` is used to define while loops in Rust. It runs the code inside it
+/// until the condition in the while loop becomes False or there the code uses
+/// `break`.
+///
+///
+/// ```rust
+/// let mut i = 0;
+/// while i > 10 {
+///     println!("i is {}", i);
+///     i += 1;
```

Comment:
Please come up with a different example where `while` is the best solution. The point of example code is usually not showing *how* to write something. The point is showing *why* you would want to write the thing. This code shows a while loop but if you were solving the problem that this code is solving then you would not want to use a while loop. That makes the example not compelling.